src/core/resource.cpp

```diff
@@ -277,6 +284,19 @@ EMGMResponse CResource::writeValue(forte::core::TNameIdentifier &paNameList, con
   return retVal;
 }
 
+void CResource::setInitialValues() {
+  CFunctionBlock::setInitialValues();
+  if(mInitList.empty()){
+    DEVLOG_ERROR("Error initializing values during reset\r\n");
+  }
```

Comment:
Is this always an error? What if there simply were no initial values set on this particular resource?